web-api/src/storage/elastic/filter.rs

```diff
@@ -498,10 +540,26 @@ impl<'a> Clauses<'a> {
                 Clause::Should(clause) => clauses.should = clause,
             }
         }
-        if !excluded.is_empty() {
-            // existing pg documents are not filtered in the query to avoid too much work for a cold
-            // path, filtering them afterwards can occasionally lead to less than k results though
```

Comment:
We exclude excluded documents because we don't want to have them into the result, that's all.

The comment seems to be based on an implicit idea about how this could maybe be implemented without using a exclusion filter during the search but past the search which I have no idea why we would explain this here. Also both implementations can lead to less then k results so I'm not sure why that is mentioned.

Reply:
iirc the comment comes from a time where returning exactly k documents was somewhat important. the comment was pointing more at that we don't want to do extra work on exlusion filtering in case the es and pg dbs are not in sync. since we are fine with less than k docs because of filtering etc, i guess the importance of that comment is not high anymore.